DataFormats/QualityControl/include/DataFormatsQualityControl/FlagReasons.h

```diff
@@ -0,0 +1,86 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#ifndef ALICEO2_ANALYSIS_FLAGREASONS
+#define ALICEO2_ANALYSIS_FLAGREASONS
+
+/// \file FlagReasons.h
+/// \brief classes keeping reasons for flagging time ranges
+/// \author Jens Wiechula, jens.wiechula@ikf.uni-frankfurt.de
+/// \author Piotr Konopka, piotr.jan.konopka@cern.ch
+
+// STL
+#include <iosfwd>
+#include <cstdint>
+
+// ROOT includes
+#include <Rtypes.h>
+
+namespace o2
+{
+namespace quality_control
+{
+
+struct FlagReasonFactory;
+
+class FlagReason
+{
+ private:
+  uint16_t mId;
+  const char* mName;
```

Comment:
Ah, I see. ROOT won't store it otherwise?
I used `const char*` so all flags with the same name could point to the same piece of memory. I will see if I can achieve it differently, while allowing for persistent storage.